lib/contentful_model/base.rb

```diff
@@ -19,53 +18,41 @@ def cache_key(*timestamp_names)
       "#{self.class.to_s.underscore}/#{self.id}-#{self.updated_at.utc.to_s(:usec)}"
     end
 
+    def hash
+      "#{sys[:content_type].id}-#{sys[:id]}".hash
+    end
+
+    def eql?(other)
+      super || other.instance_of?(self.class) && sys[:id].present? && other.sys[:id] == sys[:id]
+    end
+
     private
 
-    def define_getters
-      fields.each do |k, v|
-        if Contentful::Constants::KNOWN_LOCALES.include?(k.to_s)
-          v.keys.each do |name|
-            define_getter(name)
-          end
-        else
-          define_getter(k)
+    def override_getters
+      sys.keys.each do |name|
+        define_singleton_method name do
+          self.class.coerce_value(name, sys[name])
         end
       end
-    end
 
-    def define_getter(field_name)
-      method_name = field_name.to_s.underscore.to_sym
+      fields.keys.each do |name|
+        define_singleton_method name do
+          result = self.class.coerce_value(name, fields[name])
 
-      define_singleton_method(method_name) do
-        self.class.coerce_value(method_name, fields(default_locale)[field_name])
-      end
-    end
+          if result.is_a?(Array)
+            result.reject! { |r| r.is_a?(Contentful::Link) || (r.respond_to?(:invalid?) && r.invalid?) }
+          elsif result.is_a?(Contentful::Link)
+            result = nil
+          elsif result.respond_to?(:fields) && result.send(:fields).empty?
+            result = nil
```

Comment:
Nono, It removes `Contentful::Link` or objects that are `#invalid?` from the array of references, that means, objects that were not included due to not high enough `include` level or because they were deleted/drafted.